compiler/rustc_lint/src/early.rs

```diff
@@ -238,9 +238,10 @@ impl<'a, T: EarlyLintPass> ast_visit::Visitor<'a> for EarlyContextAndPass<'a, T>
     }
 
     fn visit_generic_param(&mut self, param: &'a ast::GenericParam) {
-        run_early_pass!(self, check_generic_param, param);
-        self.check_id(param.id);
-        ast_visit::walk_generic_param(self, param);
+        self.with_lint_attrs(param.id, &param.attrs, |cx| {
+            run_early_pass!(cx, check_generic_param, param);
+            ast_visit::walk_generic_param(cx, param);
+        });
     }
```

Comment:
Should there be similar modifications for expr field and pat fields?

Reply:
`visit_expr_field` was already there (added in #87146). I added `visit_pat_field` which fixed handling for early lints on pattern fields.